systemtest/src/test/java/io/strimzi/systemtest/mirrormaker/MirrorMaker2ST.java

```diff
@@ -893,6 +900,170 @@ void testConfigureDeploymentStrategy() {
+        JobUtils.deleteJobWithWait(NAMESPACE, sourceConsumerName);
+
+        LOGGER.info("Send {} messages to Source cluster.", MESSAGE_COUNT);
+        KafkaBasicExampleClients internalClientSourceJob = initialInternalClientSourceJob.toBuilder().withMessage("Producer B").build();
+        internalClientSourceJob.createAndWaitForReadiness(internalClientSourceJob.producerStrimzi().build());
+        ClientUtils.waitForClientSuccess(sourceProducerName, NAMESPACE, MESSAGE_COUNT);
+
+        waitForMM2OffsetIntervalMirroring(syncGroupOffsetsIntervalSeconds);
+
+        LOGGER.info("Receive {} messages from mirrored topic on Target cluster.", MESSAGE_COUNT);
+        initialInternalClientTargetJob.createAndWaitForReadiness(initialInternalClientTargetJob.consumerStrimzi().build());
+        ClientUtils.waitForClientSuccess(targetConsumerName, NAMESPACE, MESSAGE_COUNT);
+        JobUtils.deleteJobWithWait(NAMESPACE, sourceProducerName);
+        JobUtils.deleteJobWithWait(NAMESPACE, targetConsumerName);
+
+        LOGGER.info("Send 50 messages to Source cluster");
+        internalClientSourceJob = internalClientSourceJob.toBuilder().withMessageCount(50).withMessage("Producer C").build();
+        internalClientSourceJob.createAndWaitForReadiness(internalClientSourceJob.producerStrimzi().build());
+        ClientUtils.waitForClientSuccess(sourceProducerName, NAMESPACE, 50);
+        JobUtils.deleteJobWithWait(NAMESPACE, sourceProducerName);
+
+        waitForMM2OffsetIntervalMirroring(syncGroupOffsetsIntervalSeconds);
+
+        LOGGER.info("Receive 10 msgs from source cluster");
+        internalClientSourceJob = internalClientSourceJob.toBuilder().withMessageCount(10).withAdditionalConfig("max.poll.records=10").build();
+        internalClientSourceJob.createAndWaitForReadiness(internalClientSourceJob.consumerStrimzi().build());
+        ClientUtils.waitForClientSuccess(sourceConsumerName, NAMESPACE, 10);
+        JobUtils.deleteJobWithWait(NAMESPACE, sourceConsumerName);
+
+        waitForMM2OffsetIntervalMirroring(syncGroupOffsetsIntervalSeconds);
+
+        LOGGER.info("Receive 40 msgs from mirrored topic on Target cluster");
+        KafkaBasicExampleClients internalClientTargetJob = initialInternalClientTargetJob.toBuilder().withMessageCount(40).build();
+        internalClientTargetJob.createAndWaitForReadiness(internalClientTargetJob.consumerStrimzi().build());
+        ClientUtils.waitForClientSuccess(targetConsumerName, NAMESPACE, 40);
+        JobUtils.deleteJobWithWait(NAMESPACE, targetConsumerName);
+
+        LOGGER.info("There should be no more messages to read. Try to consume at least 1 message. " +
+                "This client job should fail on timeout.");
+        initialInternalClientTargetJob.createAndWaitForReadiness(initialInternalClientTargetJob.consumerStrimzi().build());
+        assertThrows(WaitException.class, () -> ClientUtils.waitForClientSuccess(targetConsumerName, NAMESPACE, 1));
```

Comment:
assert here is not needed, the test will fail even without assert in that case

Reply:
Idea is, that I expect client to fail, so this is how to check it - discussed with @im-konge

Reply:
We should keep the tests consistent so we should add this to all clients calls, or use just clients calls as we do in other tests. This exception catch doesn't look good to me.

Reply:
Added new method `ClientUtils.waitForClientTimeout()`

Reply:
I am really sorry but I miss understand the test. I thought that you are testing it in a different way. Your first solution was right. Could you please revert your changes? I am sorry for my suggestions.

Reply:
@im-konge  do you prefer `waitForClientTimeout()` in `ClientUtils` or use `assertThrow(() ->... )` ? I am fine with both solution, but in case we will use the first one, we should align the other tests with it

Reply:
If you don't want to revert it, you can keep the method and I'll change it later.

Reply:
So I believe we all agree on using method `ClientUtils.waitForClientTimeout()` (@see-quick also).
That means we can merge this PR once all tests pass. :-)